store/src/main/scala/com/improving/app/store/domain/Store.scala

```diff
@@ -119,285 +120,215 @@ object Store {
         }
       case event: StoreInfoEdited =>
         state match {
-          case _: DraftState  => DraftState(Some(event.getInfo.getEditableInfo), event.getMetaInfo)
-          case _: ReadyState  => ReadyState(event.getInfo.getInfo, event.getMetaInfo)
-          case _: OpenState   => OpenState(event.getInfo.getInfo, event.getMetaInfo)
-          case _: ClosedState => ClosedState(event.getInfo.getInfo, event.getMetaInfo)
+          case _: DraftState  => DraftState(event.info.getEditableInfo, event.metaInfo)
+          case _: ReadyState  => ReadyState(event.info.getInfo, event.metaInfo)
+          case _: OpenState   => OpenState(event.info.getInfo, event.metaInfo)
+          case _: ClosedState => ClosedState(event.info.getInfo, event.metaInfo)
           case x: StoreState  => x
         }
     }
   }
 
-  private def updateMetaInfo(metaInfo: StoreMetaInfo, lastUpdatedByOpt: Option[MemberId]): StoreMetaInfo = {
-    metaInfo.copy(lastUpdatedBy = lastUpdatedByOpt, lastUpdated = Some(Timestamp(Instant.now())))
+  private def updateMetaInfo(metaInfo: StoreMetaInfo, lastUpdatedByOpt: MemberId): StoreMetaInfo = {
+    metaInfo.copy(lastUpdatedBy = lastUpdatedByOpt, lastUpdated = Timestamp(Instant.now()))
   }
 
   private def createStore(command: CreateStore): Either[Error, StoreEvent] = {
-    val maybeValidationError = applyAllValidators[CreateStore](
-      c => required("store id", storeIdValidator)(c.storeId),
-      c => required("on behalf of", memberIdValidator)(c.onBehalfOf)
-    )(command)
-    if (maybeValidationError.isDefined) {
-      Left(maybeValidationError.get)
-    } else {
-      val newMetaInfo = StoreMetaInfo(
-        createdOn = Some(Timestamp(Instant.now())),
-        createdBy = Some(command.getOnBehalfOf)
-      )
+    val newMetaInfo = StoreMetaInfo(
+      createdOn = Timestamp(Instant.now()),
+      createdBy = command.onBehalfOf,
+      lastUpdated = Timestamp(Instant.now()),
+      lastUpdatedBy = command.onBehalfOf
+    )
 
-      Right(
-        StoreCreated(
-          storeId = command.storeId,
-          info = command.info,
-          metaInfo = Some(newMetaInfo),
-        )
+    Right(
+      StoreCreated(
+        storeId = command.storeId,
+        info = command.getInfo,
+        metaInfo = newMetaInfo
       )
-    }
+    )
   }
 
   private def makeStoreReady(
       state: DraftState,
       command: MakeStoreReady
   ): Either[Error, StoreEvent] = {
-    val draftInfo = state.info.getOrElse(EditableStoreInfo.defaultInstance)
-
-    val maybeValidationError: Option[ValidationError] = applyAllValidators[MakeStoreReady](
-      c => required("store id", storeIdValidator)(c.storeId),
-      c => required("on behalf of", memberIdValidator)(c.onBehalfOf)
-    )(command).orElse(
-      draftTransitionStoreInfoValidator(draftInfo)
-    )
-
-    if (maybeValidationError.isDefined) {
-      Left(maybeValidationError.get)
-    } else {
+    val hasErrors = draftTransitionStoreInfoValidator(state.info)
```

Comment:
Is this the solution you were talking about?

Also, `hasErrors` is a name for a boolean, not an `Option[ValidationError]`. `maybeValidationError` might be verbose but there is a reason why I went with that name when writing this pattern - it is explicit about what it is. Using names like that makes reading code quicker and easier.

Reply:
Yes, this is the solution. Will change this name and the one mentioned above